features/supplier/supplier_signs_framework_agreement.feature

```diff
@@ -0,0 +1,23 @@
+@supplier @supplier_account @gideon_test
+Feature: Supplier can sign an agreement electronically
+
+  Background:
+    Given I have the latest live or standstill framework
+    And I have a supplier user
+    And that supplier has applied to be on that framework
+    And we accepted that suppliers application to the framework
+    And that supplier is logged in
+    And that supplier has added company details
```

Comment:
Can we rename this step to be more specific (we're only updating registered name, not any other details eg DUNS number, trading status, size...). 

Or better yet, add the `set_supplier_registered_name(@supplier["id"])` function to the `And that supplier has applied to be on that framework` step, because that's the point where this data is normally updated?

Reply:
Argh this (and the comment on `api_helpers.rb` below) were left on the wrong commit. But I think the principle is the same - we should set the supplier registered name during the 'application' step (and use the existing function to do it).